src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/ApplicationInsights/ITelemetryClientFactory.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.WindowsServer.Channel.Implementation;
+
+namespace Microsoft.Azure.WebJobs.Host.Loggers
+{
+    /// <summary>
+    /// Interface for creating <see cref="TelemetryClient"/> instances.
+    /// </summary>
+    [CLSCompliant(false)]
+    public interface ITelemetryClientFactory
```

Comment:
The 'ApplicationInsightsLoggerProvider` is part of the `Microsoft.Extensions.Logging` infrastructure and I kept it internal just to limit the exposed surface area.

`ITelemetryClientFactory` is public and allows you to provide your own `TelemetryClient` for use with the provider. Script uses this to override a couple of things (notably the `SdkVersion` so the AppInsights team can track usage by WebJobs and Functions differently).

I first made the Provider public and gave it a virtual `CreateTelemetryClient()`, but you're not allowed to call virtual methods from within a constructor. I could have lazily-called the method when it was first used, but thought this was simpler and allowed everything to be initialized earlier.